python/lsst/ip/isr/isrMockLSST.py

```diff
@@ -135,63 +130,68 @@ def makeImage(self):
             if self.config.isTrimmed:
                 bbox = amp.getBBox()
             else:
-                bbox = amp.getRawDataBBox().shiftedBy(amp.getRawXYOffset())
+                bbox = amp.getRawDataBBox()
 
             ampData = exposure.image[bbox]
 
             # Sky effects in e-
             if self.config.doAddSky:
-                self.amplifierAddNoise(ampData, self.config.skyLevel, np.sqrt(self.config.skyLevel))
+                self.amplifierAddNoise(ampData, self.config.skyLevel * self.config.gain,
+                                       np.sqrt(self.config.skyLevel * self.config.gain))
 
             if self.config.doAddSource:
                 for sourceAmp, sourceFlux, sourceX, sourceY in zip(self.config.sourceAmp,
                                                                    self.config.sourceFlux,
                                                                    self.config.sourceX,
                                                                    self.config.sourceY):
                     if idx == sourceAmp:
-                        self.amplifierAddSource(ampData, sourceFlux, sourceX, sourceY)
+                        self.amplifierAddSource(ampData, sourceFlux * self.config.gain, sourceX, sourceY)
 
             # Other effects in e-
             if self.config.doAddFringe:
-                self.amplifierAddFringe(amp, ampData, np.array(self.config.fringeScale),
+                self.amplifierAddFringe(amp, ampData, np.array(self.config.fringeScale) * self.config.gain,
                                         x0=np.array(self.config.fringeX0),
                                         y0=np.array(self.config.fringeY0))
 
             if self.config.doAddFlat:
                 if ampData.getArray().sum() == 0.0:
+                    # Add noise
                     self.amplifierAddNoise(ampData, 1.0, 0.0)
-                u0 = exposure.getDimensions().getX()
-                v0 = exposure.getDimensions().getY()
+                # Multiply each amplifiers by a Gaussian centered on u0 and v0
+                u0 = exposure.getDetector().getBBox().getDimensions().getX()/2.
+                v0 = exposure.getDetector().getBBox().getDimensions().getY()/2.
                 self.amplifierMultiplyFlat(amp, ampData, self.config.flatDrop, u0=u0, v0=v0)
 
             # ISR effects
-            # 1. Add dark in e- (different from isrMock which does it in ADU)
+            # 1. Add dark in e-
```

Comment:
Add a comment that the dark current is specified in electrons per second in the config.